main.py

```diff
@@ -46,41 +48,24 @@ def generate_box():
 
 
 def run_filters(shape, noisy_shape, run_name):
-	run_and_plot_filter(
-		noisy_shape,
-		noisy_shape,
-		shape,
-		run_name,
-		"No Filter"
-	)
-	
-	run_and_plot_filter(
-		noisy_shape,
-		filters.filter_array(filters.mean_filter(4), noisy_shape),
-		shape,
-		run_name,
-		"Reference (Median Filter 4 Samples)"
-	)
-	
+	test_cases = [
+	  ("No Filter", filters.null_filter()),
```

Comment:
It would be super-cool to use a namedtuple here so that the indexing down when they're run has semantic meaning.

Something like:
```
from collections import namedtuple
TestCase = namedtuple("TestCase", ["name", "filter"])

#Then you can use:
TestCase("No Filter", filters.null_filter())

# And:

for case in test_cases:
	run_and_plot_filter(noisy_shape, filters.filter_array(case.filter, noisy_shape), shape, run_name, case.name)
```